### PR TITLE
Improve build system

### PR DESCRIPTION
**Pull request by harshula**
These changes are required by https://github.com/ACCESS-NRI/libaccessom2/pull/4

### CODE REVIEW

File: util/make_dir/make.ubuntu [hunk 1]
```diff
@@ -13,7 +13,7 @@ CHAN            = MPI1
 COUPLE          = $(OASIS_HOME)
 #
 # ARCHDIR       : directory created when compiling
-ARCHDIR         = $(COUPLE)/Linux
+ARCHDIR         = $(COUPLE)/compile_oa3-mct
```
Comment (aidanheerdegen) [inline util/make_dir/make.ubuntu:16]: Upstream they name this `compile_oa3-mct`? Not `compile_oasis3-mct`?

Comment (harshula) [reply util/make_dir/make.ubuntu:16]: Yes. https://gitlab.com/cerfacs/oasis3-mct/-/blob/OASIS3-MCT_5.0/util/make_dir/header_examples/make.ubuntu

**Review by aidanheerdegen — APPROVED** [review]
Looks a lot cleaner.

**Comment by aidanheerdegen** [comment]
I'm going to merge this so Vee can start working on the spack build.

**Pull request merged**